beacon-chain/rpc/eth/light-client/helpers.go

```diff
@@ -67,55 +48,62 @@ func createLightClientBootstrapAltair(ctx context.Context, state state.BeaconSta
 		return nil, fmt.Errorf("state slot %d not equal to latest block header slot %d", state.Slot(), latestBlockHeader.Slot)
 	}
 
-	// Prepare data
-	currentSyncCommittee, err := state.CurrentSyncCommittee()
+	// header.state_root = hash_tree_root(state)
+	stateRoot, err := state.HashTreeRoot(ctx)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not get current sync committee")
+		return nil, errors.Wrap(err, "could not get state root")
 	}
+	latestBlockHeader.StateRoot = stateRoot[:]
 
-	committee := structs.SyncCommitteeFromConsensus(currentSyncCommittee)
-
-	currentSyncCommitteeProof, err := state.CurrentSyncCommitteeProof(ctx)
+	// assert hash_tree_root(header) == hash_tree_root(block.message)
+	latestBlockHeaderRoot, err := latestBlockHeader.HashTreeRoot()
 	if err != nil {
-		return nil, errors.Wrap(err, "could not get current sync committee proof")
+		return nil, errors.Wrap(err, "could not get latest block header root")
 	}
-
-	branch := make([]string, fieldparams.NextSyncCommitteeBranchDepth)
-	for i, proof := range currentSyncCommitteeProof {
-		branch[i] = hexutil.Encode(proof)
-	}
-
-	beacon := structs.BeaconBlockHeaderFromConsensus(latestBlockHeader)
-	if beacon == nil {
-		return nil, fmt.Errorf("could not get beacon block header")
+	beaconBlockRoot, err := block.Block().HashTreeRoot()
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get block root")
 	}
-	header := &structs.LightClientHeader{
-		Beacon: beacon,
+	if !reflect.DeepEqual(latestBlockHeaderRoot, beaconBlockRoot) {
```

Comment:
Result of `HashTreeRoot()` is a `[32]byte`, which you can compare with `==` because it's not a slice but a fixed-size array.